### PR TITLE
gh #63 test profile improvements

### PR DESCRIPTION
**Pull request by ishthiyaqahmed**
Modified src/test_l1_dsAudio.c as per the yaml file

### CODE REVIEW

**Review by bhanucbp — COMMENTED** [review]
please address the review commnets

File: src/test_l1_dsAudio.c [hunk 1]
```diff
@@ -73,13 +73,13 @@
 #include <stdlib.h>
 #include "dsAudio.h"
 #include "dsAudioSettings.h"
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 2]
```diff
@@ -842,36 +981,61 @@ void test_l1_dsAudio_positive_dsGetAudioCompression(void) {
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
 	dsError_t result;
-	intptr_t handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
-	int  compression[NUM_OF_PORTS];
-	int  compression1[NUM_OF_PORTS];
+	intptr_t*  handle = NULL;
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:920]: done

File: src/test_l1_dsAudio.c [hunk 3]
```diff
@@ -842,36 +981,61 @@ void test_l1_dsAudio_positive_dsGetAudioCompression(void) {
...
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
 		// Step 03: Get the audio compression level of each port
-		result = dsGetAudioCompression(handle[i], &compression[i]);
+		result = dsGetAudioCompression(*(handle+i), (compression[0]+i));
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 4]
```diff
@@ -905,31 +1069,50 @@ void test_l1_dsAudio_negative_dsGetAudioCompression(void) {
...
+	// allocate memory for the handles
+	handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+	if(handle == NULL)
+	{
+		UT_LOG_ERROR("Failed to allcoate memory for handlers");
+		return;
+	}
+
+	//allocate memory for the compression arrays
+	for(int i = 0 ; i < (MAX_ARRAY_SIZE - 1) ; i++)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 5]
```diff
@@ -972,36 +1159,47 @@ void test_l1_dsAudio_positive_dsSetAudioCompression(void) {
...
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
 		// Step 03: Set minimum audio compression level
-		result = dsSetAudioCompression(handle[i], min_compression);
+		result = dsSetAudioCompression(*(handle+i), min_compression);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 6]
```diff
@@ -1103,31 +1313,54 @@ void test_l1_dsAudio_positive_dsGetDialogEnhancement(void) {
...
+
+	// allocate memory for the handles
+	handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+	if(handle == NULL)
+	{
+		UT_LOG_ERROR("Failed to allcoate memory for handlers");
+		return;
+	}
+
+	//allocate memory for the dialogEnhancementLevel arrays
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 7]
```diff
@@ -1103,31 +1313,54 @@ void test_l1_dsAudio_positive_dsGetDialogEnhancement(void) {
...
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 	}
 
 	// Step 03: Get the dialog enhancement levels for each port
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		result = dsGetDialogEnhancement(handle[i], &dialogEnhancementLevel[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetDialogEnhancement(*(handle+i), (dialogEnhancementLevel+i));
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 8]
```diff
@@ -1161,29 +1394,48 @@ void test_l1_dsAudio_negative_dsGetDialogEnhancement(void) {
...
+	// allocate memory for the handles
+	handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+	if(handle == NULL)
+	{
+		UT_LOG_ERROR("Failed to allcoate memory for handlers");
+		return;
+	}
+
+        //allocate memory for the dialogEnhancementLevel arrays
+        for(int i = 0 ; i < (MAX_ARRAY_SIZE - 1) ; i++)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 9]
```diff
@@ -1227,28 +1483,36 @@ void test_l1_dsAudio_positive_dsSetDialogEnhancement(void) {
...
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
 		// Step 03: Set Dialog Enhancement level - Min DE Level(0)
-		result = dsSetDialogEnhancement(handle[i], min_de_level);
+		result = dsSetDialogEnhancement(*(handle+i), min_de_level);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 10]
```diff
@@ -1358,29 +1636,51 @@ void test_l1_dsAudio_positive_dsGetDolbyVolumeMode(void) {
...
+	if(handle == NULL)
+	{
+		UT_LOG_ERROR("Failed to allcoate memory for handlers");
+		return;
+	}
 
-	bool dolbyVolumeMode[NUM_OF_PORTS];
+	//allocate memory for the dolbyVolumeMode arrays
+	for(int i = 0 ; i < (MAX_ARRAY_SIZE - 1) ; i++)
+	{
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 11]
```diff
@@ -1414,30 +1714,48 @@ void test_l1_dsAudio_negative_dsGetDolbyVolumeMode(void) {
...
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
 
-	bool dolbyVolumeMode[NUM_OF_PORTS];
+        //allocate memory for the dolbyVolumeMode arrays
+        for(int i = 0 ; i < (MAX_ARRAY_SIZE - 1) ; i++)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 12]
```diff
@@ -1479,28 +1801,39 @@ void test_l1_dsAudio_positive_dsSetDolbyVolumeMode(void) {
...
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
 		// Step 03: Set Dolby Volume Mode for each port and for each valid mode
-		result = dsSetDolbyVolumeMode(handle[i], mode);
+		result = dsSetDolbyVolumeMode(*(handle+i), mode);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 13]
```diff
@@ -1592,29 +1936,52 @@ void test_l1_dsAudio_positive_dsGetIntelligentEqualizerMode(void) {
...
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
 		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
 		// Step 03: Get the Intelligent Equalizer Mode for each port
-		result = dsGetIntelligentEqualizerMode(handle[i], &intelligentEqualizerMode[i]);
+		result = dsGetIntelligentEqualizerMode(*(handle+i), &intelligentEqualizerMode[i]);
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 14]
```diff
@@ -1592,29 +1936,52 @@ void test_l1_dsAudio_positive_dsGetIntelligentEqualizerMode(void) {
...
+        // allocate memory for the handles
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
+
+        //allocate memory for the intelligentEqualizerMode array
+        for(int i = 0 ; i < (MAX_ARRAY_SIZE - 1) ; i++)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 15]
```diff
@@ -1648,29 +2015,48 @@ void test_l1_dsAudio_negative_dsGetIntelligentEqualizerMode(void) {
...
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c]: done

File: src/test_l1_dsAudio.c [hunk 16]
```diff
@@ -9970,23 +12176,31 @@ void test_l1_dsAudio_positive_dsSetAudioMixerLevels(void) {
...
 
                 if(result == dsERR_NONE )
                 {
                         // Step 03: Call dsSetAudioMixerLevels() for each port by looping through dsAudioInput_t enum
                         for (dsAudioInput_t audioInput = dsAUDIO_INPUT_PRIMARY; audioInput < dsAUDIO_INPUT_MAX; audioInput++) {
-                                result = dsSetAudioMixerLevels(handle[i], audioInput, volume);
+                                result = dsSetAudioMixerLevels(*(handle+i), audioInput, volume);
                                 UT_ASSERT_EQUAL(result, dsERR_NONE);
                         }
                 }
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply src/test_l1_dsAudio.c:11623]: removed the null handle test , since it is not required in positive case

**Comment by ishthiyaqahmed** [comment]
closed as it is duplicate of https://github.com/rdkcentral/rdk-halif-test-device_settings/pull/114

**Pull request closed without merging**